exchangelib/properties.py

```diff
@@ -1939,7 +1924,9 @@ def get_std_and_dst(self, for_year):
         if not 0 <= len(transitions_group.transitions) <= 2:
             raise ValueError(f"Expected 0-2 transitions in transitions group {transitions_group}")
 
-        standard_period = self._get_standard_period(for_year)
+        standard_period = self._get_standard_period(transitions_group)
```

Comment:
I'd prefer to keep the original contract where `_get_standard_period` throws the exception. I'm not very fond of communicating failure via `None`.

Reply:
I totally agree. THe problem is that I don't have the `for_year` anymore in `_get_standard_period`. And it feels a bit artificial to pass it just for the exception. If you think that's fine I won't object to that though.

We can also just bubble up the `KeyError` exception, as described in your other suggestion, and deal with that at the call site? It does change the contract but at least failure is communicated explicitly.

Reply:
Ah, good catch. I agree that passing `for_year` just for the exception doesn't make sense, and the new error condition is not about the year anyway. It's not an important exception class, so we can just delete `TimezoneDefinitionInvalidForYear` entirely from this package and throw a `ValueError` here instead. Something like `ValueError(f'Transition reference {transition.to} not found in standard periods: {standard_periods_map}')`. It's just for easier debugging.

Reply:
I just pushed a change where (amongst other changes) I do pass the year, so feel free to ignore that for now. I'll add another one with a `ValueError` exception when I get the time (later today I hope).